chia/wallet/util/compute_hints.py

```diff
@@ -17,21 +20,44 @@ class HintedCoin:
     hint: Optional[bytes32]
 
 
-def compute_spend_hints_and_additions(cs: CoinSpend) -> Dict[bytes32, HintedCoin]:
-    _, result_program = cs.puzzle_reveal.run_with_cost(INFINITE_COST, cs.solution)
+def compute_spend_hints_and_additions(
+    cs: CoinSpend,
+    *,
+    max_cost: int = DEFAULT_CONSTANTS.MAX_BLOCK_COST_CLVM,
+) -> Tuple[Dict[bytes32, HintedCoin], int]:
+    cost, result_program = cs.puzzle_reveal.run_with_cost(max_cost, cs.solution)
 
     hinted_coins: Dict[bytes32, HintedCoin] = {}
     for condition in result_program.as_iter():
-        if condition.at("f").atom == ConditionOpcode.CREATE_COIN:  # It's a create coin:
-            coin: Coin = Coin(cs.coin.name(), bytes32(condition.at("rf").atom), uint64(condition.at("rrf").as_int()))
-            hint: Optional[bytes32] = None
-            if (
-                condition.at("rrr") != Program.to(None)  # There's more than two arguments
-                and condition.at("rrrf").atom is None  # The 3rd argument is a cons
-            ):
-                potential_hint: bytes = condition.at("rrrff").atom
-                if len(potential_hint) == 32:
-                    hint = bytes32(potential_hint)
-            hinted_coins[bytes32(coin.name())] = HintedCoin(coin, hint)
-
-    return hinted_coins
+        if cost > max_cost:  # pragma: no cover
+            raise ValidationError(Err.BLOCK_COST_EXCEEDS_MAX, "compute_spend_hints_and_additions() for CoinSpend")
+        atoms = condition.as_iter()
+        op = next(atoms).atom
+        if op in [
+            ConditionOpcode.AGG_SIG_PARENT,
+            ConditionOpcode.AGG_SIG_PUZZLE,
+            ConditionOpcode.AGG_SIG_AMOUNT,
+            ConditionOpcode.AGG_SIG_PUZZLE_AMOUNT,
+            ConditionOpcode.AGG_SIG_PARENT_AMOUNT,
+            ConditionOpcode.AGG_SIG_PARENT_PUZZLE,
+            ConditionOpcode.AGG_SIG_UNSAFE,
+            ConditionOpcode.AGG_SIG_ME,
+        ]:
+            cost += ConditionCost.AGG_SIG.value
```

Comment:
I see that this code seems copied from `compute_additions_with_cost()`, which also suffers from these same problems (no unit tests and no support for the hard fork). In the long run, I think it would be good if these rules (especially the cost of conditions) would be concentrated in one place, maybe even in `chia_rs`.
Perhaps `chia_rs` could provide a lower-level function that both of these can depend on.

Reply:
I experimented with this here: https://github.com/Chia-Network/chia-blockchain/pull/16661
There are some hurdles

Reply:
There's a unit test here: https://github.com/Chia-Network/chia-blockchain/blob/main/tests/wallet/test_util.py#L17 which I've updated to cover some of your other comments.

This function is a _wallet_ utility and we really just calculate/restrict the cost out of an abundance of caution.  We maybe don't even need the condition costs but in the case of offers it's nice to know it's invalid before it hits the mempool.  I don't think we want it to behave as if it knows about the hard fork because that would defeat the purpose in  offers of validating that it's mempool ready (ish).  If you have proposed simplifications I'm happy to review them in other PRs, but I don't think it really belongs here in this one.